payjoin-relay/tests/integration.rs

```diff
@@ -0,0 +1,425 @@
+        // Inside the Receiver:
+        // Enroll with relay
+        let http = reqwest::Client::builder()
+            .danger_accept_invalid_certs(true)
+            .build()
+            .expect("Failed to build reqwest http client");
+
+        // ignore error, timeout intentionally.
+        let _response = http.get(format!("{}/receive", RELAY_URL)).send().await;
+
+        // Receiver creates the payjoin URI
+        let pj_receiver_address = receiver.get_new_address(None, None).unwrap().assume_checked();
+        let amount = Amount::from_btc(1.0).unwrap();
+        let pj_uri_string = format!(
+            "{}?amount={}&pj={}",
+            pj_receiver_address.to_qr_uri(),
+            amount.to_btc(),
+            RELAY_URL,
+        );
+        let pj_uri = Uri::from_str(&pj_uri_string).unwrap();
+        let pj_uri = pj_uri.assume_checked();
+
+        // **********************
+        // Inside the V1 Sender:
+        // Create a funded PSBT (not broadcasted) to address with amount given in the pj_uri
+        let mut outputs = HashMap::with_capacity(1);
+        outputs.insert(pj_uri.address.to_string(), pj_uri.amount.unwrap());
+        debug!("outputs: {:?}", outputs);
+        let options = bitcoincore_rpc::json::WalletCreateFundedPsbtOptions {
+            lock_unspent: Some(true),
+            fee_rate: Some(payjoin::bitcoin::Amount::from_sat(2000)),
+            ..Default::default()
+        };
+        let psbt = sender
+            .wallet_create_funded_psbt(
+                &[], // inputs
+                &outputs,
+                None, // locktime
+                Some(options),
+                None,
+            )
+            .expect("failed to create PSBT")
+            .psbt;
+        let psbt = sender.wallet_process_psbt(&psbt, None, None, None).unwrap().psbt;
+        let psbt = Psbt::from_str(&psbt).unwrap();
+        debug!("Original psbt: {:#?}", psbt);
+        let (req, ctx) = RequestBuilder::from_psbt_and_uri(psbt, pj_uri)
+            .unwrap()
+            .build_with_additional_fee(
+                payjoin::bitcoin::Amount::from_sat(10000),
+                None,
+                bitcoin::FeeRate::ZERO,
+                false,
+            )
+            .unwrap();
+        log::info!("send fallback v1 to offline receiver fail");
+        let response = http
+            .post(req.url.as_str())
+            .body(req.body.clone())
+            .header("Content-Type", "text/plain")
+            .send()
+            .await
+            .expect("Failed to send request");
+        log::info!("Response: {:#?}", &response);
+        assert!(response.status().is_server_error());
+
+        // **********************
+        // Inside the Receiver:
+        tokio::task::spawn(async move {
+            let http = reqwest::Client::builder()
+                .danger_accept_invalid_certs(true)
+                .build()
+                .expect("Failed to build reqwest http client");
```

Comment:
Although `reqwest` is easier to use with `tokio` than `ureq`, it's a new dep and makes comparing this to the existing integration more burdenome